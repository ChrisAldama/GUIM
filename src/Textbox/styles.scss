@import '../styles/colors.scss';
@import '../styles/themes.scss';

.GUIMTextbox{
  -webkit-appearance: none;
  background-color: var(--white);
  border: 1px solid var(--main-gray);
  border-radius: 3px;
  display: inline-block;
  position: relative;
  outline: none;
  padding: 5px 10px;
  &:focus {
    outline: none;
    border: 1px solid var(--dark-gray);
  }
}

.GUIMTextboxBlue:focus{
    @apply --blue-theme-focus;
}
.GUIMTextboxGray:focus{
    @apply --gray-theme-focus;
}
.GUIMTextboxGreen:focus{
    @apply --green-theme-focus;
}
