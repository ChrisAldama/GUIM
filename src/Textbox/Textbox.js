import React from "react";
import "./styles.scss";

const Textbox = props => (
  <input
    className={`GUIMTextbox ${props.className} ${themes[props.color] || themes["blue"]}`}
    style={props.styles}
    type="text"
    name={props.name}
    onChange={props.onChange}
    onBlur={props.onBlur}
    onFocus={props.onFocus}
    value={props.value}
  />
);

Textbox.defaultProps = {
  name: "textbox",
  onChange: () => console.log("Textbox onChange()"),
  onBlur: () => console.log("Textbox onBlur()"),
  onFocus: () => console.log("Textbox onFocus()"),
  value: "",
  styles: {},
  className: "",
  color: "blue",
  guimInput: "textbox"
};

const themes = {
  blue: "GUIMTextboxBlue",
  gray: "GUIMTextboxGray",
  green: "GUIMTextboxGreen"
};

export default Textbox;
