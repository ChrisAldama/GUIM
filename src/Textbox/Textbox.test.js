import React from 'react';
import Textbox from './Textbox';
import renderer from 'react-test-renderer';
import { mount, shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<Textbox />', () => {
  describe('behaiviour tests', () => {
    let text = '';
    let component;
    let tree;
    let reRender;
    const makeRenderer = component =>
      () => tree = component.toJSON();

    beforeEach(() => {
      component = renderer.create(
        <Textbox
          onChange={(event) => {text = event} }
          value={text} />
      );
      reRender = makeRenderer(component);
      reRender();
    });

    it('match snapshot', () => {
      expect(tree).toMatchSnapshot();
    });

    it('write text in the input', () => {
      tree.props.onChange('random text');
      reRender();
      expect(text).toBe('random text');

      tree.props.onChange('another text');
      reRender();
      expect(text).toBe('another text');
    });
  });

  describe('dom tests', () => {
    let text = '';
    let component;

    beforeEach(() => {
      component = shallow(
        <Textbox
          name='test_textbox'
          onChange={(event) => {text = event} }
          value={text} />
      );
    });

    it('renders an input', () => {
      expect(component.name()).toBe("input");
    });

    it('name is test_textbox', () => {
      expect(component.prop('name')).toBe('test_textbox');
    });
  });
});
