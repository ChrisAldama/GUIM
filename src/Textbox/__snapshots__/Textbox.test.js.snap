// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textbox /> behaiviour tests match snapshot 1`] = `
<input
  className="GUIMTextbox  GUIMCheckboxBlue"
  name="textbox"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  style={Object {}}
  type="text"
  value=""
/>
`;
